### PR TITLE
feat: inject compilerOptions into customRenderer

### PR DESCRIPTION
**Pull request by dancon**
I fix the [issue#77](https://github.com/mrmckeb/typescript-plugin-css-modules/issues/77),

I write a new npm package [`less-plugin-aliases`](https://www.npmjs.com/package/less-plugin-aliases) and inject `compilerOptions` into the `customRenderer` that I can resolve the `@import ~path` manually.

The general use case is as below:
```json5
{
  "compilerOptions": {
    "plugins": [
      {
        "name": "typescript-plugin-css-modules",
        "options": {
          "classnameTransform": "camelCase",
          "customMatcher": "\\.less$",
          "customRenderer": "node_modules/less-plugin-aliases/lib/customRender.js"
        }
      }
    ],

    // ...
  }
}
```

ps: My english is poor, maybe there are some mistakes in README.md, pls fix it if you find ~

### CODE REVIEW

**Comment by dancon** [comment]
@mrmckeb pls take a time to review.

File: src/helpers/getDtsSnapshot.ts [hunk 1]
```diff
@@ -24,7 +25,14 @@ export const getDtsSnapshot = (
...
   }
 
-  const classes = getClasses({ css, fileName, logger, options, processor });
+  const classes = getClasses({
+    css,
+    fileName,
+    logger,
+    options,
+    processor,
+    compilerOptions: compileOpts,
```
Comment (mrmckeb) [inline src/helpers/getDtsSnapshot.ts]: Can we keep this as `compilerOptions` throughout all files? :)

Comment (dancon) [reply src/helpers/getDtsSnapshot.ts]: Sure, I will make commit tonight

**Comment by dancon** [comment]
> This looks good, thanks @dancon - only one small suggestion, and I'll also add a little more to the docs before release.
> 
> I also wonder if you could use the `customRenderer` like the below? It should work.
> 
> ```
>         "options": {
>           "classnameTransform": "camelCase",
>           "customMatcher": "\\.less$",
>           "customRenderer": "less-plugin-aliases"
>         }
> ```

It depend on the way the plugin load customRenderer, and it seems not work, becasuse you resolve the options.customRenderer base current directory [here](https://github.com/mrmckeb/typescript-plugin-css-modules/blob/409e8e607730c6595b2a83f9f2722b2a05144bca/src/index.ts#L76)

I notice that when I write the less plugin, and decided finally not to touch it , becasuse the usage like `node_module/less-plugin-aliases/lib/customRender.js` is not a big deal, and if want to load npm package directly need more code for compatibility. I can do that, but maybe next version, is that ok?

And `less-plugin-aliases` default export a plugin for less, not a customRender for this ts plugin,  so, after support load package from node_modules, the usage like below is acceptable?

```json5
"options": {
>           "classnameTransform": "camelCase",
>           "customMatcher": "\\.less$",
>           "customRenderer": "less-plugin-aliases/lib/customRender.js"
>         }
```

**Comment by dancon** [comment]
New commits happend after your first review @mrmckeb please check again:
- [chore: uniform the variable name, compilerOpts -> compilerOptions](https://github.com/mrmckeb/typescript-plugin-css-modules/pull/80/commits/786e9086361dbca7c8b176ad1f7df8c9b6880f2a)
- [feat: add support load npm package as customRenderer](feat: add support load npm package as customRenderer)

**Comment by mrmckeb** [comment]
Hi @dancon, very sorry about the slow reply.

This looks great. I'll merge today/tomorrow and add some tests for this too. I'll release by the end of the week :)

**Pull request merged**